File: gitdot-web/packages/gitdot-api/src/endpoint/auth/account/add-user-email.ts
import { z } from "zod";
import { UserEmailResource } from "../../../resource";
import type { Endpoint } from "../../endpoint";

export const AddUserEmailRequest = z.object({
  email: z.string(),
});
export type AddUserEmailRequest = z.infer<typeof AddUserEmailRequest>;

export const AddUserEmailResponse = UserEmailResource;
export type AddUserEmailResponse = z.infer<typeof AddUserEmailResponse>;

export const AddUserEmail = {
  path: "/auth/account/add-email",
  method: "POST",
  request: AddUserEmailRequest,
  response: AddUserEmailResponse,
} as const satisfies Endpoint;
export type AddUserEmail = typeof AddUserEmail;

File: gitdot-api/src/endpoint/auth/email/add_user_email.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::user::UserEmailResource};

pub struct AddUserEmail;

impl Endpoint for AddUserEmail {
    const PATH: &'static str = "/auth/email/add-email";
    const METHOD: http::Method = http::Method::POST;

    type Request = AddUserEmailRequest;
    type Response = AddUserEmailResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct AddUserEmailRequest {
    pub email: String,
}

pub type AddUserEmailResponse = UserEmailResource;
